### PR TITLE
Implemented import-meta-resolve to import ESM modules in safeImport

### PR DESCRIPTION
**Pull request by RossVertizan**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
The modifications are to the safeImport method in the file utils.ts in the package wdio-utils, to address the issue raised in #9279.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

Just some quick notes:
1. I have used the import-meta-resolve package as has been used elsewhere and discussed in #9279.
2. Since resolve throws an error if the package is not found I have used a nested try-catch, which I don't really like, but was better (IMO) than an empty catch block.  I have nesting so that I can first look on the ```import.meta.url``` path and then on the   ```localNodeModules``` path, to best replicate the existing solution.
3. I preferentially import the ESM module and if that is undefined try to load the module found by ```require.resolve```
4. I also has to add ```.js`` to the types imports in the wdio-protocols package as they were failing eslint checks.

### CODE REVIEW

File: packages/wdio-utils/src/utils.ts [hunk 1]
```diff
@@ -176,6 +178,21 @@ export async function safeImport (name: string): Promise<Services.ServicePlugin
...
+         */
+        try {
+            esmImportPath = await resolve(name, import.meta.url)
+        } catch (err: any) {
+            try {
+                esmImportPath = await resolve(name, pathToFileURL(localNodeModules).toString())
+            } catch (err: any) {
+                console.debug(`Could not determine esmImportPath using import.meta.url or localNodeModules for ${name}`)
+            }
+        }
```
Comment (christian-bromann) [inline packages/wdio-utils/src/utils.ts:194]: We don't want to do `await resolve(name, import.meta.url)` as we don't want users to load WebdriverIO dependencies, when they should load deps from their project.

Also we can remove `require.resolve` calls as `import-meta-resolve` also supports CJS modules. So I'ld suggest to replace `require.resolve` with `await resolve(...)`.

Comment (RossVertizan) [reply packages/wdio-utils/src/utils.ts:194]: Agreed

**Comment by RossVertizan** [comment]
@christian-bromann  - OK, I have updated the file in accordance with what we have discussed.  I have left your original comment.  Do we want to keep the remark about being able to link packages?

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
@RossVertizan I resolved the conflict through the UI

**Comment by christian-bromann** [comment]
I will merge and release once the tests pass.

**Pull request merged**